lib/wallet/wallet.js

```diff
@@ -3430,6 +3435,52 @@ class Wallet extends EventEmitter {
     return this.txdb.getCoinView(tx);
   }
 
+  /**
+   * Get a wallet coin viewpoint with HD paths.
+   * @param {TX} tx
+   * @returns {Promise} - Returns {@link WalletCoinView}.
+   */
+
+  async getWalletCoinView(tx) {
+    let view = tx.view;
```

Comment:
I think it would be better to have an assertion that `view` exists here because a method on it is called down the stack, then it can fail early with a nice error message instead of a `Cannot read property of undefined` error

https://github.com/handshake-org/hsd/blob/91f392ce2696cd550b1fb21668c34bf53b235ac0/lib/primitives/tx.js#L850

Reply:
I don't want the method to fail if the view is empty. How about something like this?

```
    let view = new CoinView();

    if (tx.view instanceof CoinView)
      view = tx.view;

    if (!tx.hasCoins(view))
      view = await this.txdb.getCoinView(tx);
```

Now we guard against `getWalletCoinView` passing an undefined `view` to `hasCoins`.

Reply:
Generally instances of `TX` don't have the `CoinView` right? Would it be possible to skip the `tx instanceof MTX` check?

Reply:
>Generally instances of TX don't have the CoinView right? Would it be possible to skip the tx instanceof MTX check?

Yes, `TX` objects, generally, do not carry coin views. I included the check for an instance of `MTX` to allow the reader to understand that the caller of this function can pass an `MTX`. I think this function will be cleaner if we only expect `TX` objects as the first argument and allow the caller to pass a `CoinView` as an optional, second argument. What do you think?